### PR TITLE
Validate metadata using convention

### PR DESCRIPTION
**Pull request by eweitz**
This enables metadata validation against formal conventions as part of our public CLI, `manage_study.py`.  Validation can now happen _before_ uploading, so the UX for ingest of invalid metadata files is now drastically faster.

The validation code is pulled as an external dependency from https://github.com/broadinstitute/scp-ingest-pipeline.  This aids maintainability by preventing code duplication.  See broadinstitute/scp-ingest-pipeline/pull/33 for details.

New automated tests for metadata validation mimic existing, more comprehensive upstream tests in https://github.com/broadinstitute/scp-ingest-pipeline.  These simple tests ensure our integration of that upstream code works as expected.

I've locally verified that upload and parsing for Shalek lab data for the Alexandria project still work.

This and broadinstitute/scp-ingest-pipeline/pull/33 satisfy [SCP-1900](https://broadworkbench.atlassian.net/browse/SCP-1900).

### CODE REVIEW

File: scripts/manage_study.py [hunk 1]
```diff
@@ -84,6 +93,18 @@ def login(manager=None, dry_run=False, api_base=None, verbose=False):
                       api_base=api_base)
     return(manager)
 
+def validate_metadata_file(convention_path, metadata_path):
+    with open(convention_path) as f:
+        convention = json.load(f)
+    metadata = CellMetadata(metadata_path, '', '')
+    print(f'Validating {metadata_path}')
+    format_valid = metadata.validate_format()
```
Comment (jlchang) [inline scripts/manage_study.py]: format_valid is unnecessary here. I have a note to self to clean this up in scp-ingest-pipeline

Comment (eweitz) [reply scripts/manage_study.py]: Addressed in 9ef1516.

File: scripts/manage_study.py [hunk 2]
```diff
@@ -266,116 +290,123 @@ def login(manager=None, dry_run=False, api_base=None, verbose=False):
...
-                                    email=parsed_args.email,
-                                    access=parsed_args.permission,
+if __name__ == '__main__':
+    # Login connection
+    connection = login(manager=None, dry_run=parsed_args.dry_run, api_base=api_base, verbose=verbose)
+
+    ## Handle list studies
+    if hasattr(parsed_args, "summarize_list"):
+        if verbose: print("START LIST STUDIES")
+        ret = connection.get_studies(dry_run=parsed_args.dry_run)
```
Comment (knapii-developments) [inline scripts/manage_study.py]: Why did you name this ret?

Comment (eweitz) [reply scripts/manage_study.py]: `ret` means "returned object".  That name and much of the other code in `manage_study.py` and `scp_api.py` were written by Tim.

**Pull request merged**